hudi-spark/src/main/java/org/apache/hudi/keygen/TimestampBasedKeyGenerator.java

```diff
@@ -88,6 +86,13 @@ public TimestampBasedKeyGenerator(TypedProperties config) {
 
   @Override
   public HoodieKey getKey(GenericRecord record) {
+    String recordKey = getRecordKey(record);
```

Comment:
a naive question. May I know why some key generators of them are extending from SimpleKeyGenerator and some are extending KeyGenerators. Even those extending from SimpleKeyGen, I don't see any obvious need to extend from SimpleKeyGenerator. Can you throw some light.

Reply:
I think this has been historically like this. Even I do not have any pointers around this. But we can enforce those extending from SimpleKeyGenerator to do so from KeyGenerator. Will do that.

Reply:
Usually its a good practice to use for logical reasons and not just for variables. If you think TimestampBasedKeyGenerator is an extension to Simple, it is fine. But will let you take a call. Just my 2 cents.

Reply:
@nsivabalan actually TimestampBasedKeyGenerator extending SimpleKeyGenerator has been there historically. I agree with your point, if you wish, I can refactor the code. 

Let me actually go ahead and do it.

Reply:
My bad. TimestampBasedKeyGenerator is reusing function getRecordKey() from SimpleKeyGenerator. So I guess we will keep it this way only.